sql/hive/src/test/scala/org/apache/spark/sql/catalyst/LogicalPlanToSQLSuite.scala

```diff
@@ -934,6 +934,159 @@ class LogicalPlanToSQLSuite extends SQLBuilderTest with SQLTestUtils {
     }
   }
 
+  test("mapjoin_subquery") {
+    checkSQL(
+      """
+        |SELECT subq.key1, z.value
+        |FROM (SELECT x.key as key1, x.value as value1, y.key as key2, y.value as value2
+        |      FROM src1 x JOIN src y ON (x.key = y.key)) subq
+        |JOIN srcpart z ON (subq.key1 = z.key and z.ds='2008-04-08' and z.hr=11)
+        |ORDER BY subq.key1, z.value
+      """.stripMargin,
+      "mapjoin_subquery")
+  }
+
+  test("subq2") {
```

Comment:
Sure. I actually just use the name from Hive suite for review.
I'll review this.